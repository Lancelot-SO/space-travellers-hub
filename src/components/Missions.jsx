import React from 'react';

export default function Missions() {
  return (
    <div>
      <h1>Mission</h1>
    </div>
  );
}
