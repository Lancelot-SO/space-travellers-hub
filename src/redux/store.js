import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { createLogger } from 'redux-logger';
import rocketReducer from './rocket/rocketSlice';
import missionsReducer from './missions/missionsSlice';

const logger = createLogger({
  predicate: (getState, action) => action.type !== 'SOME_ACTION',
  collapsed: true,
});

const store = configureStore({
  reducer: combineReducers({
    rockets: rocketReducer,
    missions: missionsReducer,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
