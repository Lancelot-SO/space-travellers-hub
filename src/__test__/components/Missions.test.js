import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../../components/Missions';
import store from '../../redux/store';

describe('Missions', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
