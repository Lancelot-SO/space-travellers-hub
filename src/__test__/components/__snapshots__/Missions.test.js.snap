// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly for reserved state 1`] = `
<div
  className="px-5 pr-5"
>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
